airflow-core/src/airflow/utils/sqlalchemy.py

```diff
@@ -23,12 +23,13 @@
 import logging
 from collections.abc import Generator
 from importlib import metadata
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING, TypeVar
 
 from packaging import version
 from sqlalchemy import TIMESTAMP, PickleType, event, nullsfirst
 from sqlalchemy.dialects import mysql
 from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.sql import Select
```

Comment:
This does not need to be out of the TYPE_CHECKING block if `SelectT` is defined inside  TYPE_CHECKING as well. Let’s move them both inside just in case SQLAlchemy changes where Select is declared (I believe it’s not intended to be imported directly at runtime)